js/index.js

```diff
@@ -1,42 +1,168 @@
+    'middle-img-src': 'img/mid-page-accent.jpg',
+    'services-h4':'Services',
+    'services-content': 'Services content elementum magna eros, ac posuere elvit tempus et. Suspendisse vel tempus odio, in interdutm nisi. Suspendisse eu ornare nisl. Nullam convallis augue justo, at imperdiet metus scelerisque quis.',
+    'product-h4':'Product',
+    'product-content': 'Product content elementum magna eros, ac posuere elvit tempus et. Suspendisse vel tempus odio, in interdutm nisi. Suspendisse eu ornare nisl. Nullam convallis augue justo, at imperdiet metus scelerisque quis.',
+    'vision-h4':'Vision',
+    'vision-content': 'Vision content elementum magna eros, ac posuere elvit tempus et. Suspendisse vel tempus odio, in interdutm nisi. Suspendisse eu ornare nisl. Nullam convallis augue justo, at imperdiet metus scelerisque quis.',
   },
-  "contact": {
-    "contact-h4" : "Contact",
-    "address" : "123 Way 456 Street Somewhere, USA",
-    "phone" : "1 (888) 888-8888",
-    "email" : "sales@greatidea.io",
+  'contact': {
+    'contact-h4' : 'Contact',
+    'address' : '123 Way 456 Street Somewhere, USA',
+    'phone' : '1 (888) 888-8888',
+    'email' : 'sales@greatidea.io',
   },
-  "footer": {
-    "copyright" : "Copyright Great Idea! 2018"
+  'footer': {
+    'copyright' : 'Copyright Great Idea! 2018'
   },
 };
 
 // Example: Update the img src for the logo
-let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+let logo = document.getElementById('logo-img');
+logo.setAttribute('src', siteContent['nav']['img-src'])
+
+// Navigation Menu // 
+
+// For Loop
+// const navBar = document.querySelectorAll('nav a'); // Grab all the selector of nav a
+
+// for (let i = 0; i < navBar.length; i++) { // now we want it to start from the index of 0 and loop over it.
+
+// navBar[i].textContent = Object.values(siteContent.nav)[i]; // we have to call object because its an Object.vlaue to grab the items.
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values
+
+const navLinks = document.querySelectorAll('nav a');
+navLinks.forEach(a => (a.style.color = 'green'));
+
+const services = document.getElementsByTagName('a')[0];
+services.append(siteContent['nav']['nav-item-1']);
+
+const product = document.getElementsByTagName('a')[1];
+product.append(siteContent['nav']['nav-item-2']);
+
+const vision = document.getElementsByTagName('a')[2];
+vision.append(siteContent['nav']['nav-item-3']);
+
+const features = document.getElementsByTagName('a')[3];
+features.append(siteContent['nav']['nav-item-4']);
+
+const about = document.getElementsByTagName('a')[4];
+about.append(siteContent['nav']['nav-item-5']);
+
+const contact = document.getElementsByTagName('a')[5];
+contact.append(siteContent['nav']['nav-item-6']);
+
```

Comment:
Too many queries. You already have navLinks declared, so instead of making another query for each nav link you could have used `navLinks[0].append(siteContent['nav']['nav-item-1']);`